File: problems/java_solutions/src/dk/frankandersen/P_1929_Concatenation_of_Array.java
package dk.frankandersen;

import java.util.stream.IntStream;
import java.util.Arrays;

class Solution {

    public int[] getConcatenation(int[] nums) {
        return IntStream.concat(Arrays.stream(nums), Arrays.stream(nums)).toArray();
    }
}


public class P_1929_Concatenation_of_Array {

    public void run() {
        Solution solution = new Solution();
        int[] nums = new int[] {1,2,1};
        int[] result = solution.getConcatenation(nums);
        System.out.println("Expectations:");
        System.out.println("Input: nums = [1,2,1]");
        System.out.println("Output: [1,2,1,1,2,1]");

        System.out.println("Results:");
        System.out.println(Arrays.toString(result));
    }
}

File: problems/rust_solutions/src/p_1929_concatenation_of_array.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        let mut a:Vec<i32> = Vec::from(nums.clone());
        a.extend(nums);
        a
    }
}


struct Solution {

}

pub fn run() {
    let nums:Vec<i32> = vec![1,2,1];
    let result:Vec::<i32> = Solution::get_concatenation(nums);
    println!("Expectations:");
    println!("Input: nums = [1,2,1]");
    println!("Output: [1,2,1,1,2,1]");

    println!("Results:");
    println!("{:?}", result);
}